.github/workflows/scanner-versioned-definitions-update.yaml

```diff
@@ -248,20 +248,20 @@ jobs:
   upload-definitions:
     needs:
     - build-and-run
-    if: github.event_name == 'schedule' || github.event_name == 'workflow_dispatch'
+    if: ${{ (fail() || success()) && (github.event_name == 'schedule' || github.event_name == 'workflow_dispatch') }}
     runs-on: ubuntu-latest
     steps:
     # Checkout to run ./.github/actions/download-artifact-with-retry
     - uses: actions/checkout@v4
       with:
         fetch-depth: 0
 
-    - uses: ./.github/actions/download-artifact-with-retry
+    - uses: actions/download-artifact@v4
       with:
-        # Only download vulnerability bundles.
-        # This avoids accidentally downloading nvd.zip.
+        # Vulnerability bundles are named `artifact_${{ env.SCANNER_BUNDLE_VERSION }}`.
         pattern: "artifact_*"
         path: downloaded_artifacts
+        if-no-files-found: error
```

Comment:
I am assuming the job will stop here if no bundles were uploaded, but we can also do a manual check in a separate step.